### PR TITLE
Refactor/adapt selectlist style

### PR DESCRIPTION
**Pull request by div-Leo**
**What**
This PR aims to adapt the `SelectList` component style to the given [design](https://app.zeplin.io/project/57fe438f81b870887cc71736/screen/605c7840aebc4544301abe77)

**How**
Update the color of the selected items and add the style for the disabled options.
Update the example showing the disabled options.

* feat: add disabled option style
* refactor: apply updated design
Screenshot 2021-04-20 at 16 15 01
Screenshot 2021-04-20 at 16 13 57

### CODE REVIEW

File: src/components/SelectList/SelectList.tsx [hunk 1]
```diff
@@ -131,21 +130,41 @@ const customStyles: StylesConfig = {
...
+            .filter(key => state[key])
+            .reduce((acc, style) => ({ ...acc, ...colorsByState[style] }), defaultColors);
+
+        return {
+            ...provided,
+            ...colors,
+            '&:active': {
+                ...colors
+            },
+            padding: `${state.selectProps.size === 'medium' ? '0.5rem' : '0.375rem'} 0.75rem`,
```
Comment (duranmla) [inline src/components/SelectList/SelectList.tsx:162]: While I understand that spaces should be only used on margins I wonder if paddings should have a certain guideline, padding of `0.375rem` looks arbitrary. @agonzalezUX any thoughts on this?

**Review by duranmla — CHANGES_REQUESTED** [review]
I would like to align with Designers to make sure about this decision on padding sizes.
Thanks a lot for coming to contribute @div-Leo

File: src/components/SelectList/SelectList.tsx (hunk 1 above)
Comment (agonzalezUX) [reply src/components/SelectList/SelectList.tsx:162]: we don't have paddings set in our library but they should follow the same size and structure as the margins.

Comment (agonzalezUX) [reply src/components/SelectList/SelectList.tsx:162]: here we should use or spacing (padding) 1 (0.5rem) or half of it which would be 0.25rem

Comment (duranmla) [reply src/components/SelectList/SelectList.tsx:162]: @agonzalezUX that means we need to make sure [these types of designs](https://app.zeplin.io/project/57fe438f81b870887cc71736/screen/605c7840aebc4544301abe77) get updated.

For small will be `0.25rem`, medium `0.5rem` and large `1rem`

Comment (agonzalezUX) [reply src/components/SelectList/SelectList.tsx:162]: ohh wait, my bad. Please forget what I said. I think that before use the same scale as the spacing we need to review that this is matching with the components that has padding so, for now I would not change it and would keep using what is in the zeplin file. Sorry, my fault 🙏🏼

**Review by duranmla — APPROVED** [review]
After @agonzalezUX comment then I believe we are good to go with this one.

**Comment by duranmla** [comment]
@all-contributors please add @div-Leo for his contribution on code Thanks a lot mate!

**Pull request merged**